### PR TITLE
AWS test triggered in pushes to master only

### PR DESCRIPTION
**Pull request by ggabernet**
Now that we've tested the AWS megatests it maybe we should trigger it only when making pushes to the master branch. It could be triggered too often otherwise, especially now that the pipeline is in active development.

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/viralrecon branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/viralrecon)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: .github/workflows/awstest.yml [hunk 1]
```diff
@@ -27,5 +26,5 @@ jobs:
           AWS_ACCESS_KEY_ID: ${{secrets.AWS_KEY_ID}}
           AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_KEY_SECRET}}
           TOWER_ACCESS_TOKEN: ${{secrets.TOWER_ACCESS_TOKEN}}
-        run: |
+        run: | # Submits job to AWS batch, using the pre-define job definition. Setting JVM options to "-Xms1g -Xmx4g -XX:+UseG1GC" to allow staging of larger files.
```
Comment (drpatelh) [inline .github/workflows/awstest.yml:29]:
```suggestion
        run: | # Submits job to AWS batch, using a 'nextflow-big' instance. Setting JVM options to "-XX:+UseG1GC" for more efficient garbage collection when staging remote files.

```

Comment (drpatelh) [reply .github/workflows/awstest.yml:29]: We may not need to specify `-Xms1g -Xmx4g` on the larger instance :) I ran a test for that too 🤞

File: .github/workflows/awstest.yml [hunk 2]
```diff
@@ -27,5 +26,5 @@ jobs:
           AWS_ACCESS_KEY_ID: ${{secrets.AWS_KEY_ID}}
           AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_KEY_SECRET}}
           TOWER_ACCESS_TOKEN: ${{secrets.TOWER_ACCESS_TOKEN}}
-        run: |
+        run: | # Submits job to AWS batch, using the pre-define job definition. Setting JVM options to "-Xms1g -Xmx4g -XX:+UseG1GC" to allow staging of larger files.
           aws batch submit-job --region eu-west-1 --job-name nf-core-viralrecon --job-queue 'default-8b3836e0-5eda-11ea-96e5-0a2c3f6a2a32' --job-definition nextflow-big --container-overrides '{"command": ["nf-core/viralrecon", "-r '"${GITHUB_SHA}"' -profile test_full --outdir s3://nf-core-awsmegatests/viralrecon/results-'"${GITHUB_SHA}"' -w s3://nf-core-awsmegatests/viralrecon/work-'"${GITHUB_SHA}"' -with-tower"], "environment": [{"name": "TOWER_ACCESS_TOKEN", "value": "'"$TOWER_ACCESS_TOKEN"'"}, {"name": "NXF_OPTS", "value": "-Xms1g -Xmx4g -XX:+UseG1GC"}]}'
```
Comment (drpatelh) [inline .github/workflows/awstest.yml:30]:
```suggestion
          aws batch submit-job --region eu-west-1 --job-name nf-core-viralrecon --job-queue 'default-8b3836e0-5eda-11ea-96e5-0a2c3f6a2a32' --job-definition nextflow-big --container-overrides '{"command": ["nf-core/viralrecon", "-r '"${GITHUB_SHA}"' -profile test_full --outdir s3://nf-core-awsmegatests/viralrecon/results-'"${GITHUB_SHA}"' -w s3://nf-core-awsmegatests/viralrecon/work-'"${GITHUB_SHA}"' -with-tower"], "environment": [{"name": "TOWER_ACCESS_TOKEN", "value": "'"$TOWER_ACCESS_TOKEN"'"}, {"name": "NXF_OPTS", "value": "-XX:+UseG1GC"}]}'

```

**Comment by ggabernet** [comment]
ok, great, let's see!

**Pull request merged**